marlowe-object/src/Language/Marlowe/Object/Archive.hs

```diff
@@ -0,0 +1,151 @@
+{-# LANGUAGE GADTs #-}
+{-# LANGUAGE ViewPatterns #-}
+
+-- | Provides types and functions for working with object bundle archives. This is a disk-based representation of an
+-- object bundle. The bundle is encoded as a TAR archive with each object being stored in a separate archive entry. It
+-- contains an additional manifest file that defines the order of the objects and the label of the main contract object.
+-- As such, it is a self-contained representation of a single bundled core Marlowe contract.
+--
+-- The TAR archive is compressed with gzip.
```

Comment:
Eventually, we might want to support the `.zip` format because that is less Linux-centric.

Reply:
We could just switch to that now instead